app/src/main/kotlin/pl/elpassion/report/list/adapter/AddSeparator.kt

```diff
@@ -1,24 +1,21 @@
 package pl.elpassion.report.list.adapter
 
 import com.elpassion.android.commons.recycler.ItemAdapter
-import pl.elpassion.report.list.adapter.items.*
+import pl.elpassion.report.list.adapter.items.EmptyItemAdapter
+import pl.elpassion.report.list.adapter.items.ReportItemAdapter
+import pl.elpassion.report.list.adapter.items.SeparatorItemAdapter
+import pl.elpassion.report.list.adapter.items.WeekendDayItem
 
 fun addSeparators(adapters: List<ItemAdapter<*>>) = mutableListOf<ItemAdapter<*>>().apply {
     adapters.forEachIndexed { i, itemAdapter ->
         val previousItemAdapter = adapters.getOrNull(i - 1)
-        if (areTwoItemsTheSame(itemAdapter, previousItemAdapter) ||
-                itemAdapter is DayItemAdapter && previousItemAdapter is DayNotFilledInItemAdapter ||
-                itemAdapter is DayItemAdapter && previousItemAdapter is ReportItemAdapter ||
-                itemAdapter is DayNotFilledInItemAdapter && previousItemAdapter is DayItemAdapter ||
-                itemAdapter is DayNotFilledInItemAdapter && previousItemAdapter is ReportItemAdapter) {
-
+        if (!(previousItemAdapter == null
```

Comment:
I agree, this fragment of code was begging for refactor. I did it like that because it looked more readable for me but you are right.